tests/testthat/test-renderer1-knit-print.R

```diff
@@ -134,35 +152,85 @@ test_that("clicking bottom legend adds/remove points", {
   expect_equal(get_circles(), list(10, 10))
 })
 
-plot1top$show_hide$clickElement()
-s.div <- plot1top$widget$findChildElement("class name", "selectize-input")
-s.div$clickElement()
+clickTop <- function() {
+if (remDr$browserName == "chromote") {
+  remDr$Runtime$evaluate(sprintf("div = document.getElementById('%s');div.getElementsByClassName('show_hide_selector_widgets')[0].dispatchEvent(new CustomEvent('click'));", as.character("plot1top")))
+  remDr$Runtime$evaluate(sprintf("div = document.getElementById('%s'); div.getElementsByClassName('selectize-input')[0].dispatchEvent(new CustomEvent('click'));", as.character("plot1top")))
+} else { 
+  plot1top$show_hide$clickElement()
+  s.div <- plot1top$widget$findChildElement("class name", "selectize-input")
+  s.div$clickElement()
+}
+}
+
+clickBottom <- function() {
+  if (remDr$browserName == "chromote") {
+    remDr$Runtime$evaluate(sprintf("div = document.getElementById('%s');div.getElementsByClassName('show_hide_selector_widgets')[0].dispatchEvent(new CustomEvent('click'));", as.character("plot1bottom")))
+    remDr$Runtime$evaluate(sprintf("div = document.getElementById('%s'); div.getElementsByClassName('selectize-input')[0].dispatchEvent(new CustomEvent('click'));", as.character("plot1bottom")))
+  } else { 
+    plot1top$show_hide$clickElement()
+    s.div <- plot1top$widget$findChildElement("class name", "selectize-input")
+    s.div$clickElement()
+  }
+}
+
+# Function to send a key event
+sendKey <- function(key, code, keyCode) {
+  remDr$Input$dispatchKeyEvent(type = "keyDown", key = key, code = code, windowsVirtualKeyCode = keyCode, nativeVirtualKeyCode = keyCode)
+  remDr$Input$dispatchKeyEvent(type = "keyUp", key = key, code = code, windowsVirtualKeyCode = keyCode, nativeVirtualKeyCode = keyCode)
+}
+
+sendBackspace <- function() {
+  if (remDr$browserName == "chromote") {
+    sendKey("Backspace", "Backspace", 8)
+  } else {
+    remDr$sendKeysToActiveElement(list(key="backspace"))
+  }
+  Sys.sleep(0.5)
+}
 
+sendA <- function() {
+  if (remDr$browserName == "chromote") {
+    remDr$Input$insertText(text = "a")
+    sendKey("Enter", "Enter", 13)
+  } else {
+    remDr$sendKeysToActiveElement(list("a", key="enter"))
+  }
+  Sys.sleep(0.5)
+}
+
+sendB <- function() {
+  if (remDr$browserName == "chromote") {
+    remDr$Input$insertText(text = "b")
+    sendKey("Enter", "Enter", 13)
+  } else {
+    remDr$sendKeysToActiveElement(list("b", key="enter"))
+  }
+  Sys.sleep(0.5)
+}

```

Comment:
is the only difference between these two functions the "a" and "b" ? If so then please combine them into a single function with an argument for what letter to send. (it is confusing and error-prone to have large duplicated code blocks like this)

Reply:
@tdhock I have  merged the 2 functions